src/reformatters/example/dynamical_dataset.py

```diff
@@ -0,0 +1,51 @@
+from collections.abc import Sequence
+
+from reformatters.common import validation
+from reformatters.common.dynamical_dataset import DynamicalDataset
+from reformatters.common.kubernetes import Job
+
+from .region_job import ExampleRegionJob, ExampleSourceFileCoord
+from .template_config import ExampleDataVar, ExampleTemplateConfig
+
+
+class ExampleDataset(DynamicalDataset[ExampleDataVar, ExampleSourceFileCoord]):
```

Comment:
Is this a change from our pattern to put the dataset definition in `__init__.py` (no objections, though if we do this I think we need to correct the example import to add the dataset to DYNAMICAL_DATASETS)